SpiceSharp/Algebra/Solve/Markowitz/MarkowitzDiagonal.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+
+namespace SpiceSharp.Algebra.Solve
+{
+    /// <summary>
+    /// Markowitz-based pivot strategy: diagonal search
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class MarkowitzDiagonal<T> : MarkowitzSearchStrategy<T> where T : IFormattable, IEquatable<T>
+    {
+        /// <summary>
+        /// Constants
+        /// </summary>
+        const int TiesMultiplier = 5;
+
+        /// <summary>
+        /// Find the pivot on the diagonal
+        /// </summary>
+        /// <param name="markowitz">Markowitz</param>
+        /// <param name="matrix">Matrix</param>
+        /// <param name="step">Step</param>
+        /// <returns></returns>
+        public override MatrixElement<T> FindPivot(Markowitz<T> markowitz, SparseMatrix<T> matrix, int step)
```

Comment:
@svenboulanger How about safe check for step?

Reply:
👍